LanguageFeatures/Augmentation-libraries/augmented_expression_A01_t11_lib.dart

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The exact result of an `augmented` expression depends on what is
+/// being augmented, but it generally follows the same rules as any normal
+/// identifier:
+/// - Augmenting getters: Within an augmenting getter `augmented` invokes the
+///   getter and evaluates to the return value. If augmenting a field with a
+///   getter, this will invoke the implicit getter from the augmented field.
+///
+/// @description Checks that it is a compile-time error to declare a local
+/// function named `augmented` within an augmenting getter
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+augment library 'augmented_expression_A01_t11.dart';
+
+augment String get topLevelGetter {
+  var augmented() => "x";
```

Comment:
Perhaps there's a point in using `var` because the grammar somehow allows it?

... looking ...

Nope, `⟨functionSignature⟩` is `⟨type⟩? ⟨identifier⟩ ⟨formalParameterPart⟩`, there is no way we can start with the token `'var'`.

Perhaps use `String` as the return type, so that we obviously don't get a spurious error because it's a type incorrect function?